src/utils.c

```diff
@@ -7,6 +7,19 @@
 #include <stdlib.h>
 #include <string.h>
 
+char* string_cut_maxlen(char* string, int maxlen){
+        char *buffer;
+
+        if (maxlen < 0 || strlen(string) < maxlen) {
+                buffer = string;
+        } else {
+                size_t buffer_maxlen = maxlen;
```

Comment:
Why create a new variable instead of making the type of `maxlen` be `size_t` in the function definition? And with this change, there should be no need to check if `maxlen < 0` since `size_t` is unsigned.